src/pages/my-patches/patch-card/BuildStatusIcon.tsx

```diff
@@ -1,37 +1,37 @@
-import React from "react";
+import React, { useMemo } from "react";
 import { BuildStatus } from "types/build";
 import { RunningIcon } from "pages/my-patches/patch-card/BuildStatusIcon/RunningIcon";
 import { SucceededIcon } from "pages/my-patches/patch-card/BuildStatusIcon/SucceededIcon";
 import { FailedIcon } from "pages/my-patches/patch-card/BuildStatusIcon/FailedIcon";
 import { CreatedIcon } from "pages/my-patches/patch-card/BuildStatusIcon/CreatedIcon";
-import Tooltip from "@leafygreen-ui/tooltip";
 import styled from "@emotion/styled";
+import { paths, DEFAULT_PATCH_TAB } from "constants/routes";
+import { id } from "date-fns/locale";
+import { PatchTasksQueryParams } from "types/task";
+import { useHistory } from "react-router-dom";
+import { Tooltip } from "antd";
 
 interface Props {
   status: string;
   buildVariant: string;
-  onClick: () => void;
 }
 
-export const BuildStatusIcon: React.FC<Props> = ({
-  status,
-  buildVariant,
-  onClick,
-}) => {
+export const BuildStatusIcon: React.FC<Props> = ({ status, buildVariant }) => {
+  const router = useHistory();
+  const onClick = useMemo(
+    () => () =>
+      router.push(
+        `${paths.patch}/${id}/${DEFAULT_PATCH_TAB}?${PatchTasksQueryParams.Variant}=${buildVariant}`
+      ),
+    [buildVariant, router]
+  );
```

Comment:
I have a hunch why `useMemo` is used here and it's because of the tooltip performance problems you reference in the PR summary. however i am not sure this onClick function calls for `useMemo`. When used improperly, `useCallback` and `useMemo` can actually be less performant than declaring function in component body. Essentially, there are only ever two scenarios in which `useMemo` and `useCallback` are a good idea: referential equality and expensive computations. This article from Kent C Dodds explains it all beautifully: https://kentcdodds.com/blog/usememo-and-usecallback

Reply:
that's a great article. useMemo is not needed here